src/main/java/org/gridsuite/modification/modifications/olg/OlgModification.java

```diff
@@ -0,0 +1,602 @@
+            limitsGroup.setProperty(property.name(), property.value());
+        });
+    }
+
+    protected void modifyCurrentLimits(
+            CurrentLimitsAdder limitsAdder,
+            CurrentLimits currentLimits,
+            OperationalLimitsGroupInfos.Applicability applicability) {
+        CurrentLimitsModificationInfos currentLimitsInfos = olgModifInfos.getCurrentLimits();
+        boolean hasPermanent = currentLimitsInfos.getPermanentLimit() != null;
+        if (hasPermanent) {
+            if (!(currentLimits != null && currentLimits.getPermanentLimit() == currentLimitsInfos.getPermanentLimit())) {
+                addToLogsOnSide(ModificationUtils.buildModificationReport(
+                        currentLimits != null ? currentLimits.getPermanentLimit() : Double.NaN, currentLimitsInfos.getPermanentLimit(),
+                                "Permanent limit",
+                                TypedValue.DETAIL_SEVERITY
+                        ),
+                        applicability);
+            }
+            limitsAdder.setPermanentLimit(currentLimitsInfos.getPermanentLimit());
+        } else {
+            if (currentLimits != null) {
+                limitsAdder.setPermanentLimit(currentLimits.getPermanentLimit());
+            }
+        }
+        modifyTemporaryLimits(limitsAdder, currentLimits, applicability);
+        limitsAdder.add();
+    }
+
+    private void addOlg(OperationalLimitsGroupInfos.Applicability applicability) {
+        if (applicability == EQUIPMENT || applicability == SIDE1) {
+            if (modifiedOperationalLimitsGroup1() != null) {
+                throw new PowsyblException("Cannot add " + modifiedOperationalLimitsGroup1().getId() + " operational limit group, one with the given name already exists");
+            }
+            addOlgOnASide(modifiedBranch.newOperationalLimitsGroup1(olgModifInfos.getId()), SIDE1);
+        }
+        if (applicability == EQUIPMENT || applicability == SIDE2) {
+            if (modifiedOperationalLimitsGroup2() != null) {
+                throw new PowsyblException("Cannot add " + modifiedOperationalLimitsGroup2().getId() + " operational limit group, one with the given name already exists");
+            }
+            addOlgOnASide(modifiedBranch.newOperationalLimitsGroup2(olgModifInfos.getId()), SIDE2);
+        }
+
+        if (!CollectionUtils.isEmpty(limitsReportsSide1) || !CollectionUtils.isEmpty(limitsReportsSide2)) {
+            ReportNode limitSetReport = olgsReportNode.newReportNode()
+                    .withMessageTemplate("network.modification.operationalLimitsGroupAdded")
+                    .withUntypedValue(OPERATIONAL_LIMITS_GROUP_NAME, olgModifInfos.getId())
+                    .withUntypedValue(SIDE, applicabilityToString(applicability))
+                    .withSeverity(TypedValue.INFO_SEVERITY)
+                    .add();
+            logSideNode(limitSetReport, "network.modification.operationalLimitsGroupAdded.detail", SIDE1);
```

Comment:
sonar issue 
Define a constant instead of duplicating this literal "network.modification.operationalLimitsGroupAdded.detail" 4 times.